File: src/StockGuard.Domain/Entities/Product.cs
namespace StockGuard.Domain.Entities;

public class Product
{
    public Guid Id { get; set; }
    public string Sku { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string Unit { get; set; } = "each"; // e.g. "each", "kg", "litre"
    public int ReorderLevel { get; set; }

    public Guid CategoryId { get; set; }
    public Category Category { get; set; } = null!;

    public ICollection<ProductSupplier> ProductSuppliers { get; set; } = new List<ProductSupplier>();
    public ICollection<InventoryBatch> InventoryBatches { get; set; } = new List<InventoryBatch>();
}

File: src/StockGuard.Web/src/app/core/product.ts
export interface Product {
  id: string;
  sku: string;
  name: string;
  description: string | null;
  unit: string;
  reorderLevel: number;
  categoryId: string;
  categoryName: string;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
}